### PR TITLE
Rewrite developer setup docs

### PR DESCRIPTION
**Pull request by gasman**
A rewrite of the developer setup docs to address several major failings (not mentioning Node.js or the `setup.py develop` step prominently enough, and making people set up a Vagrant environment manually when https://github.com/gasman/vagrant-wagtail-develop exists).

* State the requirement for Node.js up-front (since it's required by everyone, not just front-end devs)
* Point to vagrant-wagtail-develop instead of making people set up a Vagrant instance manually
* No need to fiddle with python paths via local.py - 'setup.py develop' handles that
* Skip instructions for virtualenvs (anyone who decides against the Vagrant route is expected to know how to use them)

### CODE REVIEW

**Comment by kaedroho** [comment]
This looks good to me! :+1:

File: docs/contributing/developing.rst [hunk 1]
```diff
@@ -1,66 +1,61 @@
 Development
 -----------
 
-Using the demo site & Vagrant
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Setting up a local copy of `the Wagtail git repository <https://github.com/torchbox/wagtail>`_ is slightly more involved than running a release package of Wagtail, as it requires `Node.js <https://nodejs.org/>`_ and NPM for building Javascript and CSS assets. (This is not required when running a release version, as the compiled assets are included in the release package.)
 
-We recommend using the `Wagtail demo site <https://github.com/torchbox/wagtaildemo/>`_ which uses Vagrant, as a basis for developing Wagtail itself.
```
Comment (tomdyson) [inline docs/contributing/developing.rst]: Should we move your script to github.com/torchbox ?

Comment (gasman) [inline docs/contributing/developing.rst]: Sure, why not :-)

Comment (gasman) [inline docs/contributing/developing.rst]: vagrant-wagtail-develop repo moved to torchbox now.

**Comment by tomdyson** [comment]
Great job, @gasman!

**Pull request merged**